api/lambdas/src/sqs-to-converter.ts

```diff
@@ -353,49 +383,76 @@ export const handler = Sentry.AWSLambda.wrapHandler(async event => {
           log
         );
 
-        await reportMemoryUsage();
-        await reportMetrics(metrics);
-        //
-      } catch (err) {
+        await cloudWatchUtils.reportMemoryUsage();
+        await cloudWatchUtils.reportMetrics(metrics);
+        //eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (err: any) {
         // If it timed-out let's just reenqueue for future processing - NOTE: the destination MUST be idempotent!
-        const count = message.attributes?.ApproximateReceiveCount;
-        if (isTimeout(err) && count <= maxTimeoutRetries) {
+        const count = message.attributes?.ApproximateReceiveCount
+          ? Number(message.attributes?.ApproximateReceiveCount)
+          : undefined;
+        const isWithinRetryRange = count == null || count <= maxTimeoutRetries;
+        const isRetryError = isTimeout(err) || isBadGateway(err);
+        if (isRetryError && isWithinRetryRange) {
           const details = `${err.code}/${err.response?.status}`;
           console.log(
             `Timed out (${details}), reenqueue (${count} of ${maxTimeoutRetries}): `,
             message
           );
-          captureMessage("Conversion timed out", {
+          capture.message("Conversion timed out", {
             extra: { message, context: lambdaName, retryCount: count },
           });
-          await reEnqueue(message);
+          await sqsUtils.reEnqueue(message);
         } else {
           console.log(
             `Error processing message: ${JSON.stringify(message)}; \n${err}: ${JSON.stringify(err)}`
           );
-          captureException(err, {
+          capture.error(err, {
             extra: { message, context: lambdaName, retryCount: count },
           });
-          await sendToDLQ(message);
+          await sqsUtils.sendToDLQ(message);
 
           if (isSidechainConnector()) {
-            await notifyApi({ cxId, patientId, status: "failed", jobId }, log);
+            await ossApi.notifyApi({ cxId, patientId, status: "failed", jobId }, log);
           }
         }
       }
     }
     console.log(`Done`);
   } catch (err) {
     console.log(`Error processing event: ${JSON.stringify(event)}; ${JSON.stringify(err)}`);
-    captureException(err, {
+    capture.error(err, {
       extra: { event, context: lambdaName, additional: "outer catch" },
     });
     throw err;
   }
 });
 
-function addExtensionToConversion(conversion, extension) {
-  const fhirBundle = conversion.fhirResource;
+function parseBody(body: unknown): EventBody {
```

Comment:
Transformation code that would otherwise pollute the logic of the main function, so isolated it